src/tools/perf/cuda/ucp_cuda_kernel.cu

```diff
@@ -248,10 +275,22 @@ private:
     ucp_perf_cuda_params m_params;
 };
 
+template<ucs_device_level_t level>
+UCS_F_DEVICE void
+ucp_perf_cuda_init_rand_state(const ucx_perf_cuda_context &ctx, curandState *rand_state) {
+    unsigned thread_index     = ucx_perf_cuda_thread_index<level>(threadIdx.x);
+    unsigned num_threads      = ucx_perf_cuda_thread_index<level>(
+            ctx.num_threads);
+    unsigned global_thread_id = blockIdx.x * num_threads + thread_index;
```

Comment:
IMO the global thread id can be calculated in a much simpler way
`threadIdx.x + blockIdx.x * blockDim.x`
so the helper function ucp_perf_cuda_init_rand_state is not needed